v23.1/fips-compliance.md

```diff
@@ -0,0 +1,355 @@
+  {{site.data.alerts.end}}
+
+This page shows how to configure FIPS-ready {{ site.data.products.core }} using Red Hat's FIPS-validated OpenSSL package. The FIPS-ready Docker image for CockroachDB comes with Red Hat's OpenSSL libraries pre-installed and configured.
+
+### Performance considerations
+
+When comparing performance of the same workload in a FIPS-ready CockroachDB runtime to a standard CockroachDB runtime, some performance degradation is expected, due to the use of the system's installed FIPS-validated OpenSSL libraries. The amount of performance impact depends upon the workload, cluster configuration, query load, and other factors.
+
+### Upgrading to a FIPS-ready CockroachDB runtime
+
+Upgrading an existing CockroachDB cluster's binaries in-place to be FIPS-ready is not supported.
+
+## Enable FIPS mode in the Linux kernel
+
+The following section provides several ways to help ensure that the FIPS object module is loaded in your system’s kernel and is used by your system's OpenSSL libraries. The FIPS object module helps to prevent the use of weak cryptographic algorithms and functions which are incompatible with FIPS 140-2.
+
+The FIPS-ready CockroachDB Docker images are based on [Red Hat's Universal Base Image 8 Docker image](https://catalog.redhat.com/software/containers/ubi8/ubi/), which comes with FIPS-validated OpenSSL libraries pre-installed with settings to comply with FIPS 140-2. To [use the FIPS-ready CockroachDB Docker image](#use-the-fips-ready-cockroachdb-docker-image), you can skip directly to that section of this page.
+
+{{site.data.alerts.callout_info}}
+A [Ubuntu Pro or Ubuntu Advantage](https://ubuntu.com/pro) subscription is required to access [FIPS-validated OpenSSL libraries and pre-configured Docker images](https://ubuntu.com/security/certifications/docs/fips). See [Enabling FIPS with the `ua` tool](https://ubuntu.com/security/certifications/docs/fips-enablement) in Ubuntu's documentation. This page does not provide specific instructions for Ubuntu.
+{{site.data.alerts.end}}
+
+A system must have FIPS mode enabled in the kernel before it can run the FIPS-ready CockroachDB binary or that starts containers that run the FIPS-ready CockroachDB Docker image. Enabling FIPS mode loads the FIPS kernel module so that the kernel itself enforces compliance with FIPS 140-2. To enable FIPS mode on Red Hat Enterprise Linux or its derivatives, refer to [Enable FIPS mode](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening) in Red Hat's documentation. To verify that FIPS mode is enabled, refer to [Verify that the kernel enforces FIPS mode](#verify-that-the-kernel-enforces-fips-mode).
```

Comment:
```suggestion
A system must have FIPS mode enabled in the kernel before it can run the FIPS-ready CockroachDB binary or the FIPS-ready CockroachDB Docker image. Enabling FIPS mode loads the FIPS kernel module so that the kernel itself enforces compliance with FIPS 140-2. To enable FIPS mode on Red Hat Enterprise Linux or its derivatives, refer to [Enable FIPS mode](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/assembly_installing-a-rhel-8-system-with-fips-mode-enabled_security-hardening) in Red Hat's documentation. To verify that FIPS mode is enabled, refer to [Verify that the kernel enforces FIPS mode](#verify-that-the-kernel-enforces-fips-mode).
```